src/scrapers/jobs/catalogo.py

```diff
@@ -0,0 +1,206 @@
+                    errors.add(s["code"])
+                    db_session.rollback()
+                else:
+                    subjects_cache.add(s["code"])
+
+            except Exception:
+                log.error("Cannot process %s", s["code"], exc_info=True)
+                errors.add(s["code"])
+
+        return len(subjects)
+
+    except Exception:
+        log.error("Cannot process search %s", base_code, exc_info=True)
+        errors.add(base_code)
+        return 0
+
+
+async def search_additional_info(code: str, db_session: Session, catalogo_session) -> None:
+    "Search code requirements and syllabus in Catalogo and save to DB"
+    log.info("Searching %s in Catalogo", code)
+
+    try:
+        data = await get_additional_info(code, session=catalogo_session)
+        syllabus = (await get_syllabus(code, session=catalogo_session)).get("syllabus")
+
+        subject: Subject = db_session.exec(
+            select(Subject).where(Subject.code == code)
+        ).one_or_none()
+        if not subject:
+            errors.add(code)
+            log.error("Discovered %s not found in DB", code)
+            return
+
+        subject.syllabus = syllabus
+        if subject.description is None:
+            subject.description = get_description(syllabus)
+
+        subject.need_all_requirements = data.get("relationship")
+        subject.restrictions = ",".join(["=".join(r) for r in data.get("restrictions", [])])
+        subject.prerequisites_raw = data.get("prerequisites_raw")
+        subject.equivalencies_raw = data.get("equivalencies_raw")
+
+        try:
+            # Set equivalencies
+            db_session.exec(
+                delete(SubjectEquivalencies).where(SubjectEquivalencies.subject_id == subject.id)
+            )
+            if "equivalencies" in data:
+                for i, group in enumerate(data["equivalencies"]):
+                    for req_code in group:
+                        req_subject_id: Subject = (
+                            db_session.exec(select(Subject).where(Subject.code == req_code))
+                            .one_or_none()
+                            .id
+                        )
+                        db_session.add(
+                            SubjectEquivalencies(
+                                subject_id=subject.id, equivalence_id=req_subject_id, group=i
+                            )
+                        )
+
+            # Set prerequisites
+            db_session.exec(
+                delete(SubjectPrerequisites).where(SubjectPrerequisites.subject_id == subject.id)
+            )
+            if "requirements" in data:
+                for i, group in enumerate(data["requirements"]):
+                    for req_code in group:
+                        is_corequisite = False
+                        if req_code[-1] == "c":
+                            is_corequisite = True
+                            req_code = req_code.strip("c")
```

Comment:
Aquí hay mucha anidación, separaría las iteraciones del `for` en una función aparte.